### PR TITLE
Build wheels on Travis-CI

### LINKED ISSUE #16: Add wheels on Pypi
Since wheel format is standard for distributing Python packages, it would be nice if we can get wheel from Pypi.

### PR DESCRIPTION
**Pull request by chezou**
Fix #16 

Currently, I release wheels on my forked GitHub repository for test purpose, but it should be modified to release on the original repo, or they can be released directly to PyPI.

- [x] Build manylinux1 wheels
- [x] Release manylinux1 wheels on forked repo
- [x] Build macOSX wheels
- [x] Release OSX wheels on forked repo
- [x] Replace target repository name

### CODE REVIEW

**Comment by chezou** [comment]
Deploy succeeded https://travis-ci.org/chezou/mecab-python3/jobs/454557257

**Comment by zackw** [comment]
@chezou Unfortunately I have to do my day job for the next several hours.  I will look at your work this evening.  Thanks for doing this, though!  I may need to make some changes but I think it will be helpful.

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*       text=auto
```
Comment (zackw) [inline .gitattributes]: If we're going to do this at all, I would prefer `* text eol=lf`.

File: .travis.yml [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
+  file_glob: true
+  file: dist/*
+  skip_cleanup: true
+  on:
+    tags: true
+    repo: SamuraiT/mecab-python3
+    condition: "$DEPLOY = true"
+  # Replace appropreate api_key
+  #api_key:
+  #  secure: q11SMpwB2uq7zhH4rreOE/5JO9gfBVBNmOD7uXSTToOAaj5qUSD/TBfI8GYELtalrVGn2O2WCrekpk4WRZ+FdH1jFUmjF/scT3j/3ReFxrgKL4++ll6H2f5pSRTmHIY4aAn4h3ZB0PN/nUqi+qxm6DgC2ptJHhCXs/zb4kFCR3izkuieGBpRndYHlthIv46A86UYw8fHLJPzSYkwYFj75bAims7Y8xTFSLC5rp4EvAQUwu8vr4iI0EgXLsnBTTAUNpoaj/3jCkF08Kp2Ecy9qywJjMtk6VBpAgseN8NDqAQpq9EiW9Gw684yqXaBr9AzIaOtkVuszTrmxjI0gh3K52CosecRgkL66RgUeBpt633ENJiOqsr4JdyWeVEIh3N75212lqHqsK1E8m5jWn0TlwDpgEMGE9GzP/YedIZzjtOVTuK9obLstZhhOAWgPQAiAVyAHe7sF9Tnc5ZR5cIMNFI64hToCQAX0EQ9pPQ1iuKc078JcxlXCJHt41fhkk2zlG48FbXioWbvMsLGV2nsXSwtylu8kw+opH9NVTx1XMxLWkE7QGDeY61NkGHf9ug8mrFQP72Bo64lcUp2uV4rQz5xItXBVhmI0XcJ9KCYYZfgn9is8dondCwNTfqqQvvYTsCZie7dglJjorAxnALZKTMtQwcbYQ5X8Jwut6f+j+I=
```
Comment (zackw) [inline .travis.yml]: Where did this api key come from? Whose identity is it borrowing?

File: .travis.yml [hunk 3]
```diff
@@ -1,25 +1,48 @@
-dist: trusty
```
Comment (zackw) [inline .travis.yml]: This diff is very confusing.  Could you please try to keep it more in the same order as it was originally?

File: .travis.yml (hunk 2 above)
Comment (chezou) [reply .travis.yml]: It's mine for test on chezou/mecab-python3. Will replace with `REPLACE-ME`.

File: .travis.yml (hunk 3 above)
Comment (chezou) [reply .travis.yml]: Will keep original order.

**Comment by chezou** [comment]
> I made some minor notes below, but the big problem with this PR is, it looks like it builds wheels that depend on system installation of the MeCab library. That's against Python packaging policy (see PEP 513, which is specifically about binary wheels for Linux, but this principle applies across the board AFAIK).

PEP 513 also allows to bundle external shared libraries, which is what `auditwheel repair` does. See  the following snipet.

```
(.venv) ➜  mecab-python3 git:(release-wheel) ✗ auditwheel show dist/mecab_python3-0.8.3-cp36-cp36m-linux_x86_64.whl       
                                                                                                                          
mecab_python3-0.8.3-cp36-cp36m-linux_x86_64.whl is consistent with the                                                    
following platform tag: "linux_x86_64".                                                                                   
                                                                                                                          
The wheel references external versioned symbols in these system-                                                          
provided shared libraries: libstdc++.so.6 with versions {'CXXABI_1.3',                                                    
'GLIBCXX_3.4'}, libgcc_s.so.1 with versions {'GCC_3.0'}, libc.so.6                                                        
with versions {'GLIBC_2.2.5'}, libm.so.6 with versions {'GLIBC_2.2.5'}                                                    
                                                                                                                          
The following external shared libraries are required by the wheel:                                                        
{                                                                                                                         
    "libc.so.6": "/lib/x86_64-linux-gnu/libc-2.23.so",                                                                    
    "libgcc_s.so.1": "/lib/x86_64-linux-gnu/libgcc_s.so.1",                                                               
    "libm.so.6": "/lib/x86_64-linux-gnu/libm-2.23.so",                                                                    
    "libmecab.so.2": "/usr/lib/libmecab.so.2.0.0",                                                                        
    "libpthread.so.0": "/lib/x86_64-linux-gnu/libpthread-2.23.so",                                                        
    "libstdc++.so.6": "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21"                                                     
}                                                                                                                         
                                                                                                                          
In order to achieve the tag platform tag "manylinux1_x86_64" the                                                          
following shared library dependencies will need to be eliminated:                                                         
                                                                                                                          
libmecab.so.2                                                                                                             

(.venv) ➜  mecab-python3 git:(release-wheel) ✗ auditwheel show repaired_wheels/mecab_python3-0.8.3-cp35-cp35m-manylinux1_x86_64.whl

mecab_python3-0.8.3-cp35-cp35m-manylinux1_x86_64.whl is consistent
with the following platform tag: "manylinux1_x86_64".

The wheel references external versioned symbols in these system-
provided shared libraries: libm.so.6 with versions {'GLIBC_2.2.5'},
libpthread.so.0 with versions {'GLIBC_2.2.5'}, libc.so.6 with versions
{'GLIBC_2.2.5', 'GLIBC_2.4'}, libgcc_s.so.1 with versions {'GCC_3.0'},
libstdc++.so.6 with versions {'CXXABI_1.3', 'GLIBCXX_3.4'}

The following external shared libraries are required by the wheel:
{
    "libc.so.6": "/lib/x86_64-linux-gnu/libc-2.23.so",
    "libgcc_s.so.1": "/lib/x86_64-linux-gnu/libgcc_s.so.1",
    "libm.so.6": "/lib/x86_64-linux-gnu/libm-2.23.so",
    "libpthread.so.0": "/lib/x86_64-linux-gnu/libpthread-2.23.so",
    "libstdc++.so.6": "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.21"
}
```

Of cource, it would be nice to have a static link, but `auditwheel repair` should bundle external shared object. so I don't think it's mandatory to use https://github.com/SamuraiT/mecab-python3/commit/dbcae6325fe445d3d0ce3b43584f9b9ffe925286 .


>  That should render both of your shell scripts unnecessary, and should also mean we don't need to download and install a dictionary during the build. Then please also look into whether using cibuildwheel would simplify things even further.

I agree to remove a dictionary. It would be nice to use cibuildwheel if we want to build windows wheels, but it is widely known to build 64bit windows MeCab DLL is tricky since it's not officialy provided. I think the method for building manylinux1 wheel is based on a [pypa repo](https://github.com/pypa/python-manylinux-demo) and it's not confusing compared to it. Since [macOS Python build on travis is not supported](https://docs.travis-ci.com/user/languages/python/), the script is a little bit tricky rather than manylinux1.

Anyway, I will try to keep original order of .travis.yml as much as I can.

**Comment by zackw** [comment]
After further research and experimentation I have determined that we really must statically link the binary wheel's extension module with libmecab.  There is no other reliable way to ensure that we have the library fixes required for #3 and similar bugs.

As such, I must decline this PR in favor of the build scripts that are now on master.  Thank you for your contribution anyway; I took a number of ideas from them, if not the exact code.

**Pull request closed without merging**